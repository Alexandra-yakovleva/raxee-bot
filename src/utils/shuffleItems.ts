export const shuffleItems = <T>(item: T[]) => {
  const res = [...item];

  for (let i = res.length - 1; i > 0; i -= 1) {
    const j = Math.floor(Math.random() * (i + 1));
    [res[i], res[j]] = [res[j], res[i]];
  }

  return res;
};
