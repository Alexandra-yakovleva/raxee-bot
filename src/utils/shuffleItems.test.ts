import { shuffleItems } from './shuffleItems';

beforeEach(() => {
  jest.spyOn(global.Math, 'random').mockReturnValue(0.3);
});

afterEach(() => {
  jest.spyOn(global.Math, 'random').mockRestore();
});

test('should shuffle array', () => {
  expect(shuffleItems(['a', 'b', 'c', 'd'])).toMatchSnapshot();
});

test('should accept empty array', () => {
  expect(shuffleItems([])).toStrictEqual([]);
});
