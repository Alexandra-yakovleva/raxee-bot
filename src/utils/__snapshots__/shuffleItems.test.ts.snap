// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should shuffle array 1`] = `
Array [
  "d",
  "c",
  "a",
  "b",
]
`;
